### ISSUE TITLE
Memberlist logger should be exported

### ISSUE THREAD

**Issue by cousine** [hashicorp/memberlist#39]
Hi there, 

I think the logger should be exported instead of the LogOutput in the configuration; I'm using a custom logger with a custom format however I must hack a custom io.Writer to have memberlist use my custom logger.

**Comment by ryanuber** [comment]
Hey @cousine, seems reasonable to use a custom logger. Instead of exporting it directly in the struct, maybe we could do like a `CreateLogger(*Config, *log.Logger)` analogous to the `Create` method or something to that effect.

**Comment by cousine** [comment]
Yeah that would be very helpful :+1: but I'd call it `CreateWithLogger` tbh :smile:

**Comment by JensRantil** [comment]
See also https://github.com/hashicorp/memberlist/issues/76.
